### ISSUE TITLE
Unclear error message in enums

### ISSUE THREAD

**Issue by jwaataja** [typetools/checker-framework#2896]
In enumerations, the enum elements are implicitly assigned, leading to an unintuitive error. Consider this enum,
```java
import org.checkerframework.checker.tainting.qual.*;

public enum C {
    @Untainted A
}
```
When I run `javac -processor TaintingChecker C.java` I get the error,
```
C.java:4: error: [assignment.type.incompatible] incompatible types in assignment.
    @Untainted A
               ^
  found   : @Tainted C
  required: @Untainted C
```
The error is that `A` is actually a field which is assigned to a default constructed instance of `C`, which is `Tainted`. However, there's no explicit assignment in the code so without knowledge of how Java implements enums this error is confusing.

When enum fields are set to invalid values, it should be clear that it's because they're implicitly being assigned to a newly constructed value. Perhaps add a different error name for invalid assignments in enums.
